subworkflows/local/prepare_genome.nf

```diff
@@ -0,0 +1,52 @@
+//
+// Uncompress and prepare reference genome files
+//
+
+// nf-core modules
+include { UNTARFILES as UNTAR_BOWTIE_INDEX } from '../../modules/nf-core/untarfiles'
+// local modules
+include { INDEX_GENOME                     } from '../../modules/local/bowtie_genome'
+
+
+workflow PREPARE_GENOME {
```

Comment:
How does this relate to when the user specifies a genome via iGenomes? Should that handling be done here as well>

Reply:
Not sure what you mean, the only params from igenomes that are used are mirtrace_species and fasta, I can move mirtrace_species handling also here

Reply:
Shouldn't we have something like:

https://github.com/nf-core/phaseimpute/blob/e7d34a4a05b962fe1fd6f609f976ed2b4b447a27/subworkflows/local/utils_nfcore_phaseimpute_pipeline/main.nf#L86-L95

where we use the fasta from igenomes if provided (and build the index) or otherwise use the `params.fasta` (and build the index if needed)

Reply:
I don't think we should, that is already done when running getGenomeAttribute('fasta') in https://github.com/nf-core/smrnaseq/blob/63da64cc3c254419c19fe5d106abfb5d18a5231a/main.nf#L31
It is the way its done in sarek https://github.com/nf-core/sarek/blob/master/main.nf and rnaseq: https://github.com/nf-core/rnaseq/blob/master/main.nf

There is no need to have the handling be done twice, it will already fetch it from igenomes if a genome is provided. If it is provided from the command line then thats the one that will be used, and if no genome is provided then it will be empty or use the default value